benchmarks/auto_benchmark.py

```diff
@@ -0,0 +1,258 @@
+import argparse
+import datetime
+import os
+import ruamel.yaml
+import shutil
+from utils import gen_model_config_json
+from utils import gen_md_report
+from utils import gen_metrics_json
+from subprocess import Popen
+
+CWD = os.getcwd()
+MODEL_JSON_CONFIG_PATH = CWD + '/model_json_config'
+BENCHMARK_TMP_PATH = '/tmp/benchmark'
+BENCHMARK_REPORT_PATH = '/tmp/ts_benchmark'
+TS_LOGS_PATH = CWD + '/logs'
+MODEL_STORE = '/tmp/model_store'
+WF_STORE = '/tmp/wf_store'
+
+class BenchmarkConfig:
+    def __init__(self, yaml_dict, skip_ts_install):
+        self.yaml_dict = yaml_dict
+        self.skip_ts_install = skip_ts_install
+        self.bm_config = {}
+        yesterday = datetime.date.today() - datetime.timedelta(days=1)
+        self.bm_config["version"] = \
+            "torchserve-nightly=={}.{}.{}".format(yesterday.year, yesterday.month, yesterday.day)
+        self.bm_config["hardware"] = 'cpu'
+
+    def ts_version(self, version):
+        for k, v in version.items():
+            if k == "branch":
+                self.bm_config["version"] = v
+            elif k == "nightly":
+                self.bm_config["version"] = "torchserve-nightly=={}".format(v)
+            elif k == "release":
+                self.bm_config["version"] = "torchserve=={}".format(v)
+            break
+
+    def models(self, model_files):
+        self.bm_config["models"] = model_files
+
+    def hardware(self, hw):
+        self.bm_config["hardware"] = hw
+
+    def metrics_cmd(self, cmd):
```

Comment:
There's code duplication between `metrics_cmd` and `report_cmd` it feels like code would be easier to understand if `cmd` was some `@dataclass` with a schema that's easy to inspect